common/coins.go

```diff
@@ -80,3 +76,57 @@ func (a EtherAmount) String() string {
 func FmtFloat(f float64) string {
 	return strconv.FormatFloat(f, 'f', -1, 64)
 }
+
+// ERC20TokenAmount represents some amount of an ERC20 token in the smallest denomination
+type ERC20TokenAmount struct {
+	amount   *big.Int
+	numUnits float64 // 10^decimals
+}
+
+// NewERC20TokenAmountFromBigInt converts some amount in the smallest token denomination into an ERC20TokenAmount.
+func NewERC20TokenAmountFromBigInt(amount *big.Int, decimals float64) *ERC20TokenAmount {
+	return &ERC20TokenAmount{
+		amount:   amount,
+		numUnits: math.Pow(10, decimals),
+	}
+}
+
+// NewERC20TokenAmount converts some amount in the smallest token denomination into an ERC20TokenAmount.
+func NewERC20TokenAmount(amount int64, decimals float64) *ERC20TokenAmount {
+	return &ERC20TokenAmount{
+		amount:   big.NewInt(amount),
+		numUnits: math.Pow(10, decimals),
+	}
+}
+
+// NewERC20TokenAmountFromDecimals converts some amount of standard token in standard format
+// to its smaller denomination.
+func NewERC20TokenAmountFromDecimals(amount float64, decimals float64) *ERC20TokenAmount {
+	numUnits := math.Pow(10, decimals)
+	amt := big.NewFloat(amount)
+	mult := big.NewFloat(numUnits)
+	res, _ := big.NewFloat(0).Mul(amt, mult).Int(nil)
```

Comment:
When writing big.Int/big.Float code, I think the preferred way to create zero values is `new(big.Int)` and `new(big.Float)`. It executes less code and is a few less characters. It is what the standard and other polished libraries use. Feel free to change it or leave it.  I won't write it on every use.

Reply:
Is this code supposed to be truncating (what it is doing) or rounding? Can we clearly document the expected behavior?

Reply:
updated, and added a comment explaining it.